### PR TITLE
[DNVB-2] Side Cart

### PR DESCRIPTION
**Pull request by flair-conor**
Initial commit

#37

### CODE REVIEW

File: assets/js/theme/_addons/global/cart-preview-addons.js [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+        event.stopImmediatePropagation();
+
+        const itemId = $(event.currentTarget).data('cart-itemid');
+        const productId = $(event.currentTarget).data('product-id');
+
+        const options = {
+            template: '../theme-components/common/cart-preview',
+        };
+
+        console.log('ey');
```
Comment (flair-duncan) [inline assets/js/theme/_addons/global/cart-preview-addons.js]: Please remove

File: assets/js/theme/_addons/global/cart-preview-addons.js [hunk 2]
```diff
@@ -0,0 +1,49 @@
+import utils from '@bigcommerce/stencil-utils';
+
+export default function () {
+    const loadingClass = 'is-loading';
+    const $cartDropdown = $('#cart-preview-dropdown');
+    const $cartLoading = $('<div class="loadingOverlay"></div>');
+
+    console.log('oy');
```
Comment (flair-duncan) [inline assets/js/theme/_addons/global/cart-preview-addons.js]: Please remove

**Comment by flair-duncan** [comment]
@flair-conor ESlint errors too, also please rebase

**Comment by flair-conor** [comment]
> Image preview is too narrow compared to design, should be same width as text element. Container is correct width but looks like the object-fit is not applied correctly:
> image

That image is taller than wide, I didn't stretch it to fit the container based on a previous conversation we had about not distorting user imagery. If you set the setting in the config to higher dimensions the issue vanishes. It is, however, not centred for some reason, so I'll get on that.

File: config.json [hunk 3]
```diff
@@ -91,7 +91,7 @@
     "productgallery_size": "500x659",
     "product_size": "800x1080",
     "productview_thumb_size": "50x50",
-    "productthumb_size": "100x100",
+    "productthumb_size": "250x250",
```
Comment (flair-duncan) [inline config.json]: Causes an issue in the main cart:
image

File: assets/scss/addons/_off-canvas-menus.scss [hunk 4]
```diff
@@ -262,3 +250,166 @@ html,
     height: 100%;
     width: 100%;
 }
+
+.previewCart {
```
Comment (flair-duncan) [inline assets/scss/addons/_off-canvas-menus.scss]: Still not in its own SCSS file

Also, this breaks the popup on the product pages:
image

File: templates/theme-components/common/cart-preview.html [hunk 5]
```diff
@@ -33,25 +43,43 @@ <h6 class="previewCartItem-name">
...
+                            <div class="previewCartItem-options">
+                                {{#each options}}
+                                    <div class="previewCartItem-option">
+                                        <span class="previewCartItem-optionName">{{name}}</span>
+                                        <span class="previewCartItem-optionValue">{{value}}</span>
+                                    </div>
+                                {{/each}}
+                            </div>
+
+                            <a class="previewCartItem-optionsLink" href="{{../../urls.cart}}#{{id}}">{{lang 'products.options'}}</a>
```
Comment (flair-duncan) [inline templates/theme-components/common/cart-preview.html]: image

Should say "Change", not "Options" `lang. 'cart.checkout.change'`

**Pull request merged**